functions-and-scope.js

```diff
@@ -70,14 +112,38 @@ const grades = [9, 8, 5, 7, 7, 4, 9, 8, 8, 3, 6, 8, 5, 6];
 // * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan opslaan?
 // Log het antwoord in de terminal.
 
+let highest = 0;
+for (let i = 0; i < grades.length; i++) {
+    if (grades[i] > highest) {
+        highest = grades[i];
+    }
+}
+
+console.log(highest);
 // ---- Verwachte uitkomst: 9
 
 
 /* 3b: Omschrijven tot een herbruikbare functie */
 // Schrijf een functie genaamd highestGrade, die een array van cijfers verwacht (zoals grades) en het hoogste cijfer teruggeeft. Gebruik hiervoor jouw antwoord van 3a.
 // Zorg ervoor dat jouw functie ook werkt als we een andere array willen checken, zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
 // Log het antwoord in de terminal.
+function highestGrade(gradesArray) {
+
+    let highest = 0;
```

Comment:
Het is altijd beter om te beginnen met de eerste waarde van de array. Beetje flauw bij dit voorbeeld, maar stel dat je een array hebt van `[-2, -5, -1, -3]`. Als nu deze array aan jouw functie mee zou geven, dan komt daar `0` uit, terwijl dat cijfer helemaal niet in de array voorkomt.

```suggestion
    let highest = gradesArray[0];
```